packages/repository-tests/src/crud/relations/acceptance/belongs-to.relation.acceptance.ts

```diff
@@ -106,6 +106,26 @@ export function belongsToRelationAcceptance(
       expect(result).to.be.undefined();
     });
 
+    it('queries entities with null foreign key', async () => {
+      const customer = await customerRepo.create({
+        name: 'Thor',
+      });
+
+      // order with customer relation
+      await orderRepo.create({
+        customerId: customer.id,
+        description: 'Take Out',
+      });
+
+      // order without customer relation
+      await orderRepo.create({
+        description: 'Dine in',
+      });
+
+      let results = await orderRepo.find({include: [{relation: 'customer'}]});
+      expect(results.length).to.equal(2);
```

Comment:
What are we trying to fix in the PR? I didn't see the code changes, and new added test doesn't seem to test the `null` fk.

Reply:
Hi @agnes512, I moved in with only the tests first to check all the datasources that fail for this condition. I am adding the code changes now.

Reply:
I should have opened a draft PR, sorry about the false alarm.

Reply:
@agnes512 the new tests do check for null fk , because the order 'Dine in' has no customer Id.

Reply:
Oh haha I see. Sorry that I got the notification and just reviewed it. 

And I have a few questions. I checked other tests in this file, the test in [#L 97](https://github.com/strongloop/loopback-next/pull/4795/files#diff-c77e7e4e7301ea16d5acce2d1252dcd1R97) checks the case that source instance does not have the foreign key. Why is that one passing before the changes? Is the bug caused by the inclusion? If so, I think file `belongs-to.inclusion-resolver.relation.acceptance.ts` fits this test more. From my understanding, `order.customer(order.id)` is different from `order.find({include customer}`, because the later one uses the inclusion resolver, where the problem lies in. It'd also be better if we can check content of `result` to see if there is only one instance has the related model.

Reply:
@agnes512 the other test passes because it directly queries the target relation. You are right, I will move this test to the inclusion resolver tests. it suites better there.